lib/src/http/uncased.rs

```diff
@@ -172,6 +189,20 @@ impl<'s> Uncased<'s> {
         self.string.into_owned()
     }
 
+    /// Converts `self` into a `Box<UncasedStr>`.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use rocket::http::uncased::Uncased;
+    ///
+    /// let boxed = Uncased::new("Content-Type").into_boxed_uncased_str();
+    /// assert_eq!(&*boxed, "content-type");
+    /// ```
+    pub fn into_boxed_uncased_str(self) -> Box<UncasedStr> {
+        unsafe { transmute(self.string.into_owned().into_boxed_str()) }
```

Comment:
Likewise with the `transmute`; we should be able to do this without the `transmute`.